client/dashboard/components/summary-button-list/index.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+	Card,
+	CardHeader,
+	__experimentalVStack as VStack,
+	__experimentalText as Text,
+} from '@wordpress/components';
+import clsx from 'clsx';
+import { isValidElement, cloneElement, Children, ReactElement } from 'react';
+import { SummaryButtonListProps } from './types';
+import './style.scss';
+
+/**
+ * The SummaryButtonList is a utility component that wraps multiple SummaryButton instances
+ * along with an optional section header. It provides consistent layout, spacing, and
+ * grouping behavior for presenting a collection of summary actions or options.
+ *
+ * This component ensures visual coherence and structural alignment when multiple
+ * SummaryButton elements are displayed together. It is intended for use in contexts
+ * where a list of summarised items or actions needs to be grouped under a shared
+ * label or heading.
+ */
+export function SummaryButtonList( {
+	title,
+	description,
+	density = 'medium',
+	children,
+}: SummaryButtonListProps ) {
+	const isMediumDensity = density === 'medium';
+	// Clone children and override their density prop.
+	const clonedChildren = Children.map( children, ( child ) => {
+		if ( isValidElement( child ) ) {
+			return cloneElement( child as ReactElement< { density?: string } >, { density } );
+		}
+		return child;
+	} );
+	const header = (
+		<VStack spacing={ 4 }>
+			<h3 className="client-dashboard-components-summary-button-list__heading">{ title }</h3>
+			{ description && <Text variant="muted">{ description }</Text> }
+		</VStack>
+	);
+	const className = clsx(
+		'client-dashboard-components-summary-button-list',
+		`has-density-${ density }`
+	);
+	if ( isMediumDensity ) {
+		return (
+			<Card className={ className }>
+				<CardHeader>{ header }</CardHeader>
+				<VStack
+					spacing="1px"
+					className="client-dashboard-components-summary-button-list__children-container"
```

Comment:
Those long classnames are giving me headaches 😅 Is this some remnant of abiding `wpcalypso/jsx-classname-namespace`? Should we keep them shorter?